spring-vault-core/src/main/java/org/springframework/vault/core/lease/domain/Lease.java

```diff
@@ -84,6 +86,14 @@ public boolean isRenewable() {
 		return renewable;
 	}
 
+	/**
+	 *
+	 * @return {@literal true} if the lease represents a rotating generic secret.
+	 */
+	public boolean isRotatingGenericLease() {
```

Comment:
Rotation is a Spring Vault concept, not necessarily bound to a `Lease`. I'd propose to rather introduce a method `boolean hasLeaseId()` instead. Rotation and generic secret backend aren't really tied to a lease.